### ISSUE TITLE
Greylist issue with cPanel

### ISSUE THREAD

**Issue by jeff-h** [mail-in-a-box/mailinabox#1445]
My MIAB install has problems receiving email from cPanel servers. Specifically due to the greylisting feature, I think.

I have two cPanel servers, one with Site5 and another with Namecheap. My email accounts with Site5 are *never* received by my MIAB. They don't bounce back, they simply vanish. Unfortunately, Site5 don't give me access to any logs so it's hard to debug at that end. I couldn't even find mention in my MIAB log of those emails being received. It's like they vanish.

On the Namecheap side, my first test never got through (subsequent emails do get through). It generated two log entries in cPanel's `Email > Track Delivery` section:

1. `SMTP error from remote mail server after RCPT TO:<jeff@my.miab.domain>: 450 4.2.0 <jeff@ my.miab.domain >: Recipient address rejected: Greylisted, see http://postgrey.schweikert.ch/help/my.miab.domain.html`

...followed 11 minutes later by:

2. `retry time not reached`

My second test (with a different Namecheap cPanel email account), got the same `450` response from MIAB's greylisting feature. cPanel then retried sending that email 48 minutes later and it was delivered perfectly.

**In summary it sounds to me like MIAB does not like a redelivery so soon as 11 minutes, but 48 minutes is fine.**

Could anyone more knowledgeable than me tell me where the config and/or logs for the greylisting system are in MIAB? And whether there's any credibility in my theory above regarding the redelivery time?

Obviously "ditch cPanel" is a good solution to this :) However, with it being so massively popular I'd like to be sure my MIAB receives email from anyone with cPanel just fine, which I don't think it does right now.

**Comment by JoshData** [comment]
I think that's right and that it's expected behavior when the sender's retry times don't line up well with the greylisting settings. For more see https://postgrey.schweikert.ch/. Postgrey whitelists senders once they pass greylisting once, so this problem usually only occurs for first-time senders.

**Comment by jeff-h** [comment]
Hmm — I dug further and it appears MIAB configures postgrey to a delay of 180 seconds:

* https://github.com/mail-in-a-box/mailinabox/commit/11546b97bb409a2fbfcc8cae8ae2e08be18ca356
* https://github.com/mail-in-a-box/mailinabox/blob/master/setup/mail-postfix.sh#L213

My understanding on that is that if the sender retries again after 180 seconds, MIAB should accept the email. That appears not to have happened in my above test, instead sending the "retry time not reached" error.

I'm not eager to stick my head in the sand on this, when I know emails are not being accepted. Unfortunately I'm finding email issue debugging pretty difficult, especially when I only have root access to MIAB. Any suggestions?

Anyone else have access to cPanel email accounts and can confirm this problem? I've had problems accepting 2 out of 2 cPanel servers' email.

**Comment by alento-group** [comment]
I sent two separate emails from an account on a cPanel from BuyShared.  They were sent to two completely different installs of MiaB.
Both were greylisted - as expected.
And both were delivered.  One 18 minutes later and the other 31 minutes later.  Both of these delivery times seemed long to me, but the end result was that they both were delivered.

**Comment by jeff-h** [comment]
Thanks @alento-group. I have come to suspect that my cPanel emails are failing because I am expecting too much sophistication of my cPanel installs. For example, I chose to use Cloudflare for the DNS of my cPanel domains, meaning setting SPF and DKIM records is no longer "click this button in cPanel", and are hence not set on mine. Also, I have configured a self-signed cert on the cPanel where email doesn't get through at all, so I am wondering if MIAB knows the sender's cert is self-signed and rejects mail based on that?

All said and done, I think this issue is due to the way I have (mis)configured cPanel, and not to do with MIAB at all.
